File: Data Structures & Algorithms/palindromic-substrings/submission-4.java
class Solution {
    int solve(int left,int right,String s){
        int count=0;
        while(left>=0 && right<s.length() && s.charAt(left)==s.charAt(right)){
            left--;
            right++;
            count++;
        }
        return count;
    }
    public int countSubstrings(String s) {
        int res=0;
        for(int i=0;i<s.length();i++){
            res+=solve(i,i,s);
            res+=solve(i,i+1,s);
        }
        return res;
    }
}

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-4.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
      unordered_set<char> set(s.begin(),s.end());
      int ans=0;
      for(char c:set){
      int left=0,count=0;
      for(int right=0;right<s.size();right++){
        if(s[right]==c) count++;
        while((right-left+1)-count>k){
            if(s[left]==c) count--;
            left++;
        }
        ans=max(ans,right-left+1);
      }  
      }
      return ans;
    }
};
